tests/test_protocols/test_generator.py

```diff
@@ -359,7 +355,7 @@ def test_generated_protocol_end_to_end(self):
         agent_1_handler = Agent1Handler(
             skill_context=SkillContext(aea_1.context), name="fake_skill"
         )
-        resources_1.handler_registry.register(
+        resources_1._handler_registry.register(
```

Comment:
For now I don't think we should add a public API to add handlers or behaviours without a skill. It is not clean given current architecture and comes with potential dangerous side effects (e.g. certain APIs breaking which require skill configs etc present)

Reply:
yes, agreed.